python/mxnet/optimizer.py

```diff
@@ -332,7 +333,8 @@ def create_state(self, index, weight):
         if self.momentum == 0.0:
             return None
         else:
-            return zeros(weight.shape, weight.context, dtype=weight.dtype)
+            return sparse_zeros(weight.storage_type, weight.shape,
```

Comment:
its better to add a storage_type argument to zeros and make it output sparse arrays accordingly.